### PR TITLE
[feat][clock] Add digital clock output command

### PR DESCRIPTION
**Pull request by jidicula**
**Why this change was necessary**
The command for setting menubar digital clock output was missing from
the list.

**What this change does**
Adds config information for the digital clock time and date format
with a few examples.

**Any side-effects?**
None.

Note that I haven't tested this on Big Sur. Once this testing is complete, I can add Big Sur in the `versions` field.

**Additional context/notes/links**

Resolves yannbertrand/macos-defaults#41

See also: https://github.com/herrbischoff/awesome-macos-command-line#set-menu-bar-clock-output-format

### CODE REVIEW

**Comment by jidicula** [comment]
Did I do something wrong with the filename for the video recording? The [deploy check](https://app.netlify.com/sites/macos-defaults/deploys/5f79fbb023e6f9000902a306) failed with this message:

```
12:45:42 PM: Error: ENOENT: no such file or directory, copyfile '../../images/menubar/DateFormat/EEE d MMM HH:mm:ss.mp4' -> 'docs//menubar/menubar-DateFormat-EEE d MMM HH:mm:ss.mp4'
```

**Comment by yannbertrand** [comment]
That means the video files don't exist 😄

If you want to generate them, you'll have to create [a script to do it programmatically](https://github.com/yannbertrand/macos-defaults/tree/master/record#readme). You can take a look at the [menubar FlashDateSeparators](https://github.com/yannbertrand/macos-defaults/blob/master/record/menubar/FlashDateSeparators/true.js) scripts that should be very similar.

If you don't want to do it, I'll add it for you 😉

**Comment by jidicula** [comment]
> That means the video files don't exist 😄
> 
> If you want to generate them, you'll have to create [a script to do it programmatically](https://github.com/yannbertrand/macos-defaults/tree/master/record#readme). You can take a look at the [menubar FlashDateSeparators](https://github.com/yannbertrand/macos-defaults/blob/master/record/menubar/FlashDateSeparators/true.js) scripts that should be very similar.
> 
> If you don't want to do it, I'll add it for you 😉

Sure, I can take a crack at it 🙂. Quick question though, when I name the js script, it makes sense to name it after the config's value that has been applied – if I follow this convention, how should I handle the spaces in the filename? Any pitfalls I should be aware of, or is it "safe" to just name the script `EEE d MMM HH:mm:ss.js` for example.

**Comment by yannbertrand** [comment]
> > That means the video files don't exist 😄
> > If you want to generate them, you'll have to create [a script to do it programmatically](https://github.com/yannbertrand/macos-defaults/tree/master/record#readme). You can take a look at the [menubar FlashDateSeparators](https://github.com/yannbertrand/macos-defaults/blob/master/record/menubar/FlashDateSeparators/true.js) scripts that should be very similar.
> > If you don't want to do it, I'll add it for you 😉
> 
> Sure, I can take a crack at it 🙂. Quick question though, when I name the js script, it makes sense to name it after the config's value that has been applied – if I follow this convention, how should I handle the spaces in the filename? Any pitfalls I should be aware of, or is it "safe" to just name the script `EEE d MMM HH:mm:ss.js` for example.

Cool. I'd say use underscores `_` instead of spaces ` ` in the file name.

**Comment by jidicula** [comment]
Ok, I *thought* I set up the recording scripts correctly but the Netlify build still failed 😦 – where am I going wrong?

**Comment by yannbertrand** [comment]
You did good! Afaik, Netlify or other servers can't run these scripts themselves. In fact I run them on my Mac and upload the videos inside the repo.

So let me run them locally and I'll push them afterwards!

File: defaults.yml [hunk 1]
```diff
@@ -346,6 +346,37 @@ categories:
...
             text: The time separator flashes every second.
         after: killall SystemUIServer
+      - key: DateFormat
+        domain: com.apple.menuextra.clock
+        title: Set menubar digital clock format
+        description:
+          This setting configures the time and date format for the menubar
+          digital clock. Default value depends on your locale settings. `ss`
+          for seconds, `HH` for 24-hour clock, `EEE` for 3-letter day of the
+          week, `d MMM` for day of the month and 3-letter month.
```
Comment (yannbertrand) [inline defaults.yml:356]:
```suggestion
          This setting configures the time and date format for the menubar
          digital clock. Default value depends on your locale settings. `ss`
          for seconds, `HH` for 24-hour clock, `EEE` for 3-letter day of the
          week, `d MMM` for day of the month and 3-letter month.
          These commands could have a different result depending on the Language & Region parameters.
```

File: defaults.yml [hunk 2]
```diff
@@ -346,6 +346,37 @@ categories:
...
+              width: 727
+              height: 40
+            text: "Example time output: Thu 9:46:18"
+          - value: "EEE HH:mm:ss"
+            video:
+              filename: "EEE_HH:mm:ss.mp4"
+              width: 727
+              height: 40
+            text: "Example time output: Thu 21:46:18"
+        versions: [Catalina]
```
Comment (yannbertrand) [inline defaults.yml:378]:
```suggestion
        versions: [Big Sur, Catalina]
```

File: defaults.yml [hunk 3]
```diff
@@ -346,6 +346,37 @@ categories:
...
+        after: killall SystemUIServer
```
Comment (yannbertrand) [inline defaults.yml:379]: killall SystemUIServer is not required
```suggestion
```

File: defaults.yml [hunk 4]
```diff
@@ -346,6 +346,37 @@ categories:
...
+          week, `d MMM` for day of the month and 3-letter month.
+        param:
+          type: string
+        examples:
+          - value: "EEE d MMM HH:mm:ss"
+            video:
+              filename: "EEE_d_MMM_HH:mm:ss.mp4"
+              width: 727
+              height: 40
+            text: "Example time output: Thu 18 Aug 21:46:18"
```
Comment (yannbertrand) [inline defaults.yml:365]:
```suggestion
            text: Thu 18 Aug 21:46:18
```

File: defaults.yml [hunk 5]
```diff
@@ -346,6 +346,37 @@ categories:
...
+          - value: "EEE h:mm:ss"
+            video:
+              filename: "EEE_h:mm:ss.mp4"
+              width: 727
+              height: 40
+            text: "Example time output: Thu 9:46:18"
```
Comment (yannbertrand) [inline defaults.yml:371]:
```suggestion
            text: Thu 9:46:18
```

File: defaults.yml [hunk 6]
```diff
@@ -346,6 +346,37 @@ categories:
...
+          - value: "EEE HH:mm:ss"
+            video:
+              filename: "EEE_HH:mm:ss.mp4"
+              width: 727
+              height: 40
+            text: "Example time output: Thu 21:46:18"
```
Comment (yannbertrand) [inline defaults.yml:377]:
```suggestion
            text: Thu 21:46:18
```

**Review by yannbertrand — CHANGES_REQUESTED** [review]
Just a few notes.

Will probably add some examples later, and I need to fix something on the code snippets before merging.

Thanks again!

**Comment by yannbertrand** [comment]
Here are the video files, if you can push them on your branch! 😄 

[DateFormat.zip](https://github.com/yannbertrand/macos-defaults/files/5324144/DateFormat.zip)

**Comment by jidicula** [comment]
> Here are the video files, if you can push them on your branch! 😄
> 
> [DateFormat.zip](https://github.com/yannbertrand/macos-defaults/files/5324144/DateFormat.zip)

Did the filenames get changed? I see slashes instead of colons.

**Comment by yannbertrand** [comment]
> > Here are the video files, if you can push them on your branch! 😄
> > [DateFormat.zip](https://github.com/yannbertrand/macos-defaults/files/5324144/DateFormat.zip)
> 
> Did the filenames get changed? I see slashes instead of colons.

I'm not sure why it does that, it should be colons.

**Comment by jidicula** [comment]
Looks like macOS doesn't like colons in filenames. I'll use dots instead.
image

File: defaults.yml [hunk 7]
```diff
@@ -346,6 +346,37 @@ categories:
...
+        description:
+          This setting configures the time and date format for the menubar
+          digital clock. Default value depends on your locale settings. `ss`
+          for seconds, `HH` for 24-hour clock, `EEE` for 3-letter day of the
+          week, `d MMM` for day of the month and 3-letter month.
+          These commands could have a different result depending on the Language & Region parameters.
+        param:
+          type: string
+        examples:
+          - value: "EEE d MMM HH:mm:ss"
```
Comment (yannbertrand) [inline defaults.yml:361]: Fix snippet copy/paste
```suggestion
          - value: '"EEE d MMM HH:mm:ss"'
```

File: defaults.yml [hunk 8]
```diff
@@ -346,6 +346,37 @@ categories:
...
+            video:
+              filename: "EEE_d_MMM_HH:mm:ss.mp4"
+              width: 727
+              height: 40
+            text: Thu 18 Aug 21:46:18
+          - value: "EEE h:mm:ss"
```
Comment (yannbertrand) [inline defaults.yml:367]:
```suggestion
          - value: '"EEE h:mm:ss"'
```

File: defaults.yml [hunk 9]
```diff
@@ -346,6 +346,37 @@ categories:
...
+            video:
+              filename: "EEE_h:mm:ss.mp4"
+              width: 727
+              height: 40
+            text: Thu 9:46:18
+          - value: "EEE HH:mm:ss"
```
Comment (yannbertrand) [inline defaults.yml:373]:
```suggestion
          - value: '"EEE HH:mm:ss"'
```

**Review by yannbertrand — CHANGES_REQUESTED** [review]
Fix website snippet copy/paste

**Comment by jidicula** [comment]
Looks like some issues with asset size:
image

Deploy looks fine otherwise though 😁

**Comment by yannbertrand** [comment]
> Looks like some issues with asset size:
> image
> 
> Deploy looks fine otherwise though 😁

😂 You got them all. That Lighthouse performance measure is unstable, it doesn't even run on the page you've created. Relaunching the build fixed it 😉

**Review by yannbertrand — APPROVED** [review]

**Comment by yannbertrand** [comment]
Thank you so much again!

**Pull request merged**